lib/gpu/render_pass.cc

```diff
@@ -134,8 +140,27 @@ RenderPass::GetOrCreatePipeline() {
   render_pipeline_->BindToPipelineDescriptor(*context.GetShaderLibrary(),
                                              pipeline_desc);
 
-  auto pipeline =
-      context.GetPipelineLibrary()->GetPipeline(pipeline_desc).Get();
+  std::shared_ptr<impeller::Pipeline<impeller::PipelineDescriptor>> pipeline;
+
+  if (context.GetBackendType() == impeller::Context::BackendType::kOpenGLES) {
```

Comment:
Similar question here, are we misssing some thread handling in the gles shader library

Reply:
I think the reactor was WAI here. When attempting to create a pipeline from the UI thread, the pipeline creation task was getting stored in the reactor because GL calls need to be done on the raster thread. The problem is I need the pipeline right away, and so I need to post a raster task.

It's excessive to post a raster task every time we do a pipeline lookup like this. We only need to do it when we know a new pipeline needs to be created, but the API is lacking at the moment.

Reply:
Do you actually need the pipeline object, or could we store the pipeline future?

Alternatively, consider making this API future returning so we don't end up with weird thread pinning issues.

Reply:
Yeah we need the pipeline object immediately. This happens during `RenderPass.draw`. I don't think we should make it async. But we could prevent this raster task in cases where a new pipeline doesn't actually need to be created.

Reply:
I changed this to not post a raster task unless we know a new pipeline actually needs to be created.